src/Utils/arm/request.ts

```diff
@@ -93,7 +95,7 @@ export async function armRequest<T>({
   }
 
   const operationStatusUrl = response.headers && response.headers.get("location");
-  if (operationStatusUrl) {
+  if (shouldPollOperationStatus && operationStatusUrl) {
```

Comment:
>shouldPollOperationStatus [](start = 6, length = 25)

if someone passes shouldpoll to true but the RP api doesnt come back with location, do you want to throw an error?

Reply:
if the RP doesn't return a polling url, it means the request has gone through and there is no need of polling. So in that case we should just return the result and keep it backwards compatible with other APIs using armRequest().